chemprop/cli/train.py

```diff
@@ -876,17 +887,17 @@ def train_model(
                 preds = np.split(preds, 4, axis=1)[0]
 
             evaluate_and_save_predictions(
-                preds, test_loader, model.metrics[:-1], model_output_dir, args
+                preds, test_loader, model.metrics[:-1], model_output_dir, args, columns
             )
 
         best_model_path = checkpointing.best_model_path
         model = model.__class__.load_from_checkpoint(best_model_path)
         p_model = model_output_dir / "best.pt"
-        save_model(p_model, model)
+        save_model(p_model, model, args.target_cols)
```

Comment:
```suggestion
        save_model(p_model, model, args.target_columns)
```
```suggestion
        save_model(p_model, model, target_cols)
```
depending on if you take my suggestion to not overwrite `args.target_columns`